Atomic mass update issue #1057

makoziol0 opened the pull request:
Replaced the old mass file with the newer version from #1057 and redid some tests to match.

pshriwise committed:
Updating test to comply with the pymoab switch.

pshriwise committed:
Allowing indexing of tag data directly on volume elements in the mesh.

pshriwise committed:
Removing iMesh-based calls and replacing with direct calls to the pyne.mesh tag interface.

pshriwise committed:
Removing IMeshTag include in alara and test_alara to avoid importing another mesh type when it isn't completely necessary.

pshriwise committed:
Updating CI docker images to test pymoab build.

pshriwise committed:
Correcting build line in ci file.

pshriwise committed:
Updating PYTAPS check in test files.

pshriwise committed:
Corrections to get/set checks for indexing type.

pshriwise committed:
Re-adding a valid assumption about the tag size.

pshriwise committed:
Corrections to the dagmc module for use with pymoab.

pshriwise committed:
Replacing iMesh calls in fluka tests with pyne Mesh calls.

pshriwise committed:
Updating mcnp iMesh calls.

pshriwise committed:
Correcting the index iterator in pyne mesh.

pshriwise committed:
Moving mcnp module calls to pyne.mesh module calls (mostly). And all calls in test_mcnp to exterior mesh modules calls.

pshriwise committed:
Creating ways to work with tagging mesh rootSets without introducing that concept into PyNE too deeply. Adding a convenience method for getting all tags on a PyNE mesh.

pshriwise committed:
Removing some additional print statements in the mcnp module.

pshriwise committed:
Updating how existing tags are added as IMeshTags and improving the identification of the mesh for root set tagging.

pshriwise committed:
Updating an old iMesh call in r2s.

pshriwise committed:
Updating iMesh calls to pymoab.

pshriwise committed:
Updating HAVE_PYTAPS check in test_mcnp.

pshriwise committed:
Adding option for writing Mesh class without materials. Also updating index value retrieval from mesh tag.

pshriwise committed:
Updating mesh tag calls in vr module.

pshriwise committed:
Changes to get things working with Python3.

pshriwise committed:
Adding comment about explicit type-cast.

pshriwise committed:
Updating docker image to pull

pshriwise committed:
Circle CI is lame

pshriwise committed:
Grasping at oh-so-few straws.

pshriwise committed:
Changes to circle ci config.

pshriwise committed:
Trying something else.

pshriwise committed:
Relying on HAVE_PYTAPS in dagmc test now.

pshriwise committed:
Updating HAVE_PYTAPS to HAVE_PYMOAB

pshriwise committed:
Enabling usage and testing of PyNE without PyMOAB.

pshriwise committed:
Adding test for Python2 without full mesh support.

pshriwise committed:
Fixing install sans PyMOAB lines.

pshriwise committed:
Creating separate build with pymoab

pshriwise committed:
Detecting if DAGMC is found properly. Ignoring LD_LIBRARY_PATH environment to follow PyNE protocol.

pshriwise committed:
Updating tests s.t. mesh tests wil run without dagmc.

pshriwise committed:
Removing MOAB/DAGMC from Python3 install on CI.

pshriwise committed:
Cleanup of modules and tests external to the mesh module.

pshriwise committed:
Removing mention of iMesh in mesh module.

pshriwise committed:
Renaming the IMeshTag class.

pshriwise committed:
Updates to expand_tags for pymoab usage.

pshriwise committed:
Updating some style in the mesh module.

pshriwise committed:
Some further style changes.

pshriwise committed:
Populating structured mesh coordinates in a more elegant/efficient manner.

pshriwise committed:
More style updates for default args.

pshriwise committed:
Removing comment about find_boxes.

pshriwise committed:
Some more cleanup for readability.

pshriwise committed:
More cleanup for readability.

pshriwise committed:
Updating all instances of 'imesh' in source files.

pshriwise committed:
Updating all instances of 'imesh' in test files.

pshriwise committed:
Correcting test function name.

ngranda committed:
fixing mesh used to comply with pyMOAB

gonuke committed:
Don't make MOAB require DAGMC

gonuke committed:
Only Cythonize dagmc if DAGMC_FOUND

gonuke committed:
only link dagmc if DAGMC_FOUND

bam241 committed:
add longer timeout time

bam241 committed:
empty

bam241 committed:
clean

gonuke committed:
update API for mesh tally call to save

kkiesling committed:
check for dagmc in header

gonuke committed:
streamline the call to save()

kkiesling committed:
check for dagmc when running disctretize_geom

kkiesling committed:
checks for dagmc for mat assignments

gonuke committed:
allow 2 minutes for downloading files during testing

gonuke committed:
a skeleton of a delete_tag method

kkiesling committed:
updated partisn tests

kkiesling committed:
correctly accounts for new mat_assigns argument

kkiesling committed:
moved where dagmc is checked in tests

kkiesling committed:
wrong optional argument

kkiesling committed:
checks for dagmc for isotropic volume source

kkiesling committed:
remove unnecessary print statement

ngranda committed:
delete dagmc dependency on r2s.py

ngranda committed:
change tests to comply with changes to r2s.py

kkiesling committed:
tests for manual inputs

kkiesling committed:
remove unused test

gonuke committed:
Add a delete_tag method that removes all tag references to properly delete tags

gonuke committed:
Add some robustness

gonuke committed:
add test for robustness

gonuke committed:
Add delete methods to Tags and call explicitly.

gonuke committed:
cleanup __delete__ method

Commit:
fixing deleting tags in r2s script

bam241 committed:
some update

bam241 committed:
some typo/improvements

bam241 committed:
improvement

ngranda committed:
comment out dagmc dependency

ngranda committed:
r2s_test changes

zxkjack123 committed:
update dependencies

ngranda committed:
change cell_fracs to structed array

kkiesling committed:
correctly sets up structured array for tests

ngranda committed:
clean up

zxkjack123 committed:
update docs for source sampling

ngranda committed:
correct description error

ngranda committed:
test cleanup

bam241 committed:
some new changes

bam241 committed:
some new changes

bam241 committed:
formating

Commit:
changes to get and destroy tags

zxkjack123 committed:
simplify the variable float_tag_size

zxkjack123 committed:
Update tutorial/07-mesh-basics.ipynb

zxkjack123 committed:
compatibility of configparser

zxkjack123 committed:
fix tag, cell_fracs problem

zxkjack123 committed:
fix sub-voxel r2s cell_fracs tag

gonuke committed:
Update docs/theorymanual/source_sampling.rst

Co-Authored-By: zxkjack123 <zxkjack123@163.com>

gonuke committed:
Update docs/usersguide/source_sampling.rst

Co-Authored-By: zxkjack123 <zxkjack123@163.com>

zxkjack123 committed:
add list about mesh volume element

zxkjack123 committed:
update

gonuke committed:
Update news/pymoab_cleanup.rst

Co-Authored-By: bam241 <bam241@users.noreply.github.com>

ngranda committed:
Update news/pymoab_cleanup.rst

Co-Authored-By: bam241 <bam241@users.noreply.github.com>

bam241 committed:
other version

bam241 committed:
more destroy

bam241 committed:
get_tag, delete_tag

zxkjack123 committed:
clean up

makoziol0 commented:
Doesn't seem like the automated test is getting the new mass file data loaded in. Although when I do it manually on my computer it works fine. Advice? @katyhuff

gonuke commented on cmake/ReplicatePythonSourceTree.cmake:
```diff
@@ -12,5 +12,6 @@ file(COPY setup.py setup_sub.py pyne DESTINATION "${CMAKE_ARGV3}"
                     PATTERN "*.pyi"
                     PATTERN "*.pxd"
                     PATTERN "*.mas12"
```
If we delete the 2012 file, we can probably remove this line.

makoziol0 replied:
Removed.

gonuke reviewed (commented):
Just one little comment.

As for failing tests - I wonder if the continuous integration tests are not incorporating this new data and are using old data in the docker image - maybe @bam241 or @kkiesling knows and/or can confirm?

makeclean commented:
one should also run atomc_data_gen.py and commit the new cpp files

makoziol0 commented:
> one should also run atomc_data_gen.py and commit the new cpp files

@makeclean 
The atomic_data.cpp and .h files from atomicgen.py are in the gitignore file so I think they are excluded on purpose.

gonuke commented:
I agree with @makoziol0 on this - the CircleCI reports that new `atomic_data.h` and `atomic_data.cpp` files were generated by `atomicgen.py` during build.

I poked around a little further in the build/test process.
1. we do not invoke `nuc_data_make` during CI - we use a pre-existing nuclear data file
2. the pre-existing nuclear data file has the old atomic mass data

I think we need to use a special procedure to merge this and also update the nuclear data file in the CI image.

zxkjack123 committed:
check pymoab import message

gonuke committed:
More options for MAOB and DAGMC

1. Each of MOAB and DAGMC can be added with no path, in which case some default path is searched.
2. Each of MOAB and DAGMC can be added with specific paths
3. If not specified at all, no search for MOAB or DAGMC will be done

gonuke committed:
Fix CMake syntax

gonuke committed:
small variable name change for clarity

gonuke committed:
streamline a loop

gonuke committed:
fixing extra text from rebase

zxkjack123 committed:
Fix typo 1

Co-Authored-By: gonuke <paul.wilson@wisc.edu>

gonuke committed:
Commit a set of typos in comments and documentation

Co-Authored-By: gonuke <paul.wilson@wisc.edu>

gonuke committed:
remove file that is now redundant

zxkjack123 committed:
fix typo in warning message

Missed this in prior typo cleanup batch

Co-Authored-By: gonuke <paul.wilson@wisc.edu>

gonuke committed:
removing extra line

gonuke committed:
A few more typos

gonuke committed:
correct mesh handling with move of dagmc dependence in r2s

gonuke committed:
Adding a comment for clarity

gonuke committed:
Add flux_tag to interface for resolve_mesh

gonuke committed:
Need to pass output_material in resolve_mesh

gonuke committed:
a little more cleanup

gonuke committed:
correct missing value

gonuke committed:
Fix off-by-1 error

gonuke committed:
improve consistency of importing HAVE_PYMOAB and pyne.mesh components

gonuke committed:
consistency in asserts

gonuke committed:
Adding comment about use of str()

gonuke committed:
Remove unused variable

gonuke committed:
Code reuse and test iterable lists in the roght order

gonuke committed:
add some detail to porting tips in news

gonuke committed:
missing scope on function calls

gonuke committed:
need to pass more data to collect_iterables

gonuke committed:
Add a docstring for StatMesh

gonuke committed:
import pymoab.core as mb_core for clarity

gonuke committed:
silly syntax error

gonuke committed:
remove mesh.save() and turn back on tests for source sampling

gonuke committed:
add specific details on error tag name

gonuke committed:
Change iteritems() to items()

According to [1] this is correct syntax in both Py2 & Py3 but inefficient in 2.

gonuke committed:
move location of SkipTest

zxkjack123 committed:
add check pymoab for a test

zxkjack123 committed:
add doc string for resolve_mesh

zxkjack123 committed:
more check

bam241 committed:
working python script

bam241 committed:
adding --all option

bam241 committed:
push

bam241 committed:
no bash script required

bam241 committed:
doc string + import cleanup

bam241 committed:
cleaning

bam241 committed:
ortho

bam241 committed:
new ci config

bam241 committed:
fix pyne-deps name

bam241 committed:
Test

bam241 committed:
Test

bam241 committed:
ci version bump

bam241 committed:
test

bam241 committed:
test

bam241 committed:
merge build and test

bam241 committed:
cleaning

bam241 committed:
new flag default

bam241 committed:
fixing typo

bam241 committed:
missing space

bam241 committed:
flag typo

bam241 committed:
flag typo

bam241 committed:
test no timeout

bam241 committed:
flagss fix

bam241 committed:
fix

bam241 committed:
re add python as parameter

bam241 committed:
STATIC: no pymoab

bam241 committed:
various fix

bam241 committed:
mising space

bam241 committed:
split build/test --1

bam241 committed:
bad name

bam241 committed:
typo

bam241 committed:
missing -

bam241 committed:
missing arg in run_test

bam241 committed:
SHA fix

bam241 committed:
fix

bam241 committed:
some fix and dependancies

bam241 committed:
missing :

bam241 committed:
better with " \?\?

bam241 committed:
better with " \?\?

bam241 committed:
an other attempt

bam241 committed:
fixing path

bam241 committed:
adding comments

zxkjack123 committed:
fix unstructured r2s tag delete

bam241 committed:
test

bam241 committed:
new test

bam241 committed:
cleaning, fixing ortho

bam241 committed:
add executor

bam241 committed:
removing moab form dagmc build

bam241 committed:
dagmc build required MOAB flag

bam241 committed:
add python3 option for docker generation, update docker image names and update circle accordingly

bam241 committed:
lower case variable

bam241 committed:
putting back the apt package...

bam241 committed:
fixing SHA

bam241 committed:
fixing SHA - attempt2

bam241 committed:
attempt 3

bam241 committed:
attempt 4

bam241 committed:
dagmc alone does not make to much sense

bam241 committed:
dagmc alone does not make to much sense

bam241 committed:
dagmc alone does not make to much sense

bam241 committed:
add some comments

gonuke committed:
Update docker/make_pyne_docker_image.py

Co-Authored-By: bam241 <bam241@users.noreply.github.com>

zxkjack123 committed:
Update scripts/r2s.py

missed this name change in prior commit

Co-Authored-By: gonuke <paul.wilson@wisc.edu>

gonuke committed:
Add CircleCI changes

Add information about changes to CircleCI

gonuke committed:
Apply suggestions from initial code review by @katyhuff

Thanks for this quick review @katyhuff

Co-Authored-By: gonuke <paul.wilson@wisc.edu>

katyhuff committed:
Apply additional suggestions from initial code review by @katyhuff

Missed these ones in collapsed code - thanks @katyhuff

Co-Authored-By: gonuke <paul.wilson@wisc.edu>

ljacobson64 committed:
Add include guards around measure.h for amalgamated pyne

zxkjack123 committed:
add error message when e_bounds is missing

zxkjack123 committed:
add news file

zxkjack123 committed:
adopt ljacobson64's suggestion

bam241 committed:
enforcing c++11

bam241 committed:
remove extra blank line

gonuke committed:
fix problem with FileNotFound exception

gonuke committed:
better variable name

gonuke committed:
initialize in constructor

gonuke committed:
fix syntax error around string addition

gonuke committed:
Streamline and cleanup FileNotFound

zxkjack123 committed:
format using autopep8

zxkjack123 committed:
change a multiline comment

zxkjack123 committed:
fix __future__ line position

zxkjack123 committed:
fix more __future__ positons

zxkjack123 committed:
change tests import sequence

zxkjack123 committed:
fix a indent error

zxkjack123 committed:
fix valuesview not support index

zxkjack123 committed:
change import sequence

zxkjack123 committed:
fix import sequence in test_source_sampling

zxkjack123 committed:
fix import error in test_variancereduction

zxkjack123 committed:
use new sampler for both default and subvoxel r2s

gonuke committed:
Update docker/make_pyne_docker_image.py

Co-Authored-By: bam241 <bam241@users.noreply.github.com>

zxkjack123 committed:
add news file

zxkjack123 committed:
add r2s examples and test functions

zxkjack123 committed:
add news file

zxkjack123 committed:
remove difference between DEFAULT and SUBVOXEL

zxkjack123 committed:
remove comment lines

zxkjack123 committed:
remove data for ALARA from r2s_examples

zxkjack123 committed:
change a file name

makeclean committed:
Fixed bug where expand elements under the presence of no data was inventing nuclides

makeclean committed:
Corrected test case

makeclean committed:
Test needs to performed in stages, so as to ensure full reload of the CPP data

makeclean committed:
Copypasta random text

makoziol0 committed:
Merge branch 'develop' into massupdate

katyhuff commented:
Ok -- so ... I think we should open an issue describing the problem that is blocking this PR merge (continuous integration tests are not incorporating the new data introduced by the PR and are instead using old data in the docker image ). The student who submitted this PR is no longer working with me. Perhaps @Mongwell @aruzzo could take a look at this and:

- [x] check to see if any existing PyNE issue would address these CI failures (continuous integration tests are not incorporating the new data introduced by the PR and are instead using old data in the docker image )
- [x] create an issue describing how CI should be updated (e.g. should we run nuc_data_make in CI? or, perhaps only do so if a particular PR flag is present... ?)
- [x] take some steps to try solving this issue... so that then this pr can pass the tests and be merged...

gonuke commented:
Should we have a meeting to talk through this?  It seems like something that would benefit from having a lot of heads together to untangle some of the dependencies?

katyhuff commented:
Let me see if I can grasp it all by meeting with @aruzzo  and @Mongwell here at Illinois, then maybe we can get back to this once it's more coherent.

gonuke commented:
Sure thing... it’s seems that they have a pretty good handle on it, but others (read: me) need to be brought up to speed to help merge it all

gonuke commented:
replaced by #1281
